src/components/Tasks.js

```diff
@@ -15,74 +16,65 @@ export const Tasks = () => {
 
     const [tempItem, setTempItem] = useState(initial);
 
-    const getItems = () => {
-        fetch(uri)
-        .then(res => res.json())
-        .then(
-            result => {
-                setIsLoaded(true);
-                setTodoItems(result);
-            }, error => {
-                setIsLoaded(true);
-                setError(error);
-            }
-        );
+    const getItems = async () => {
+        //setIsLoaded(false);
+        try {
+            const result = await getItemsApi();
+            setTodoItems(result);
+        } catch (e) {
+            setError(`Error getting items: ${e}`);
+        }
+        setIsLoaded(true);
     };
 
-    const addItem = () => {
+    const addItem = async () => {
         const item = {
             isComplete: false,
             name: newTodoName.trim()
         };
 
-        fetch(uri, {
-            method: 'POST',
-            headers: {
-                'Accept': 'application/json',
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(item)
-        })
-        .then(response => response.json())
-        .then(() => {
+        try {
+            await addItemApi(item);
             getItems();
+            // setTodoItems([...todoItems, {
+            //     id: item.id,
+            //     ...item
+            // }])
```

Comment:
there a reason why u can't just do the commented out code instead of making a nother api call (the less calls the better)

Reply:
```suggestion
setTodoItems([...todoItems, {
    ...result
 }])
```